packages/rum-core/src/common/patching/xhr-patch.js

```diff
@@ -89,21 +92,39 @@ export function patchXMLHttpRequest(callback) {
     if (listener) {
       oriRemoveListener.call(target, READY_STATE_CHANGE, listener)
     }
-    const newListener = (target[XHR_LISTENER] = () => {
-      if (target.readyState === target.DONE) {
-        // sometimes on some browsers XMLHttpRequest will fire onreadystatechange with
-        // readyState=4 multiple times, so we need to check task state here
-        if (
-          !data.aborted &&
-          XMLHttpRequest[XHR_SCHEDULED] &&
-          task.state === SCHEDULE
-        ) {
-          invokeTask(task)
+
+    let earlierEvent
+    const newListener = (target[XHR_LISTENER] = ({ type }) => {
+      /**
+       * In certain frameworks (e.g. angular/http) the http request is aborted
+       * as soon as it completes, and that causes the state of the XHR to change.
+       * See https://github.com/angular/angular/issues/33822 for more.
+       */
+      if (earlierEvent) {
+        if (earlierEvent != type) {
```

Comment:
The check makes sure that both the `load` and `readystatechange` are fired. Since the order is not guaranteed we can't assume the load event would be the last. even though in all the cases I tested the load was, this is not part of the spec.

Reply:
The order should be guaranteed here since its our custom listeners and all listeners are called in the order they are registered.

Reply:
The order of execution is not part of the spec, therefore it depends on the specific implementation and different implementation do it differently. I think the bottom line is if something is not part of the spec we can't assume it will be guaranteed.